packages/custom_lint/lib/src/output.dart

```diff
@@ -0,0 +1,269 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+import 'package:cli_util/cli_logging.dart';
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+/// The maximum length of any of the existing severity labels.
+const int _severityWidth = 7;
+
+/// The number of spaces needed to indent follow-on lines (the body) under the
+/// message. The width left for the severity label plus the separator width.
+const int _bodyIndentWidth = _severityWidth + 3;
+
+final String _bodyIndent = ' ' * _bodyIndentWidth;
+
+/// is used as the usageLineLength.
+int? get _dartdevUsageLineLength =>
+    stdout.hasTerminal ? stdout.terminalColumns : null;
```

Comment:
I don't understand how the function name and its implementation are related

Reply:
Just taken from the Dart analyzer - it is the max line length in a terminal before a line break happens, but only if stdout is connected a terminal.